js/util/database.js

```diff
@@ -79,4 +80,10 @@ db.open().then(function () {
 
 Dexie.Promise.on('error', function (error) {
   console.warn('database error occured', error)
+
+  if (!dbAlertShown) {
```

Comment:
Instead of triggering this on every error, I think we should instead check whether or not the error message has the "Internal error opening backing store" string in it.

Reply:
The problem is that this callback is called more than once with the same internal error message, so checking only the message shows the alert 6 times on a fresh open (with only the tour tab opened)

Reply:
OK, I didn't realize that, maybe check both the message and ```dbAlertShown```?

Reply:
That makes sense, I'm just wondering if there's other types of errors that can occur. If some other kind of error (unrelated to the multiple-instance issue) occurs, we probabably don't want to show the error message even if it is the first error to happen.

Reply:
I see, then the whole logic will be changed. Will do it tomorrow 😃

Reply:
I'm not sure you would have to change much, I think you could just do this:
```
if (!dbAlertShown && error.toString().indexOf("..
```

But I haven't actually tested this either, so maybe it's more complicated than that.

Reply:
I put the check at the beginning because if another error would be triggered, the flag would be set to `true` and the error message would never be checked, because the `!dbAlertShown` would be `false` so there would be no reason to evaluate the second part of the `and` operation.